fdbserver/DataDistribution.actor.h

```diff
@@ -80,32 +86,68 @@ FDB_DECLARE_BOOLEAN_PARAM(WantNewServers);
 FDB_DECLARE_BOOLEAN_PARAM(WantTrueBest);
 FDB_DECLARE_BOOLEAN_PARAM(PreferLowerUtilization);
 FDB_DECLARE_BOOLEAN_PARAM(TeamMustHaveShards);
+FDB_DECLARE_BOOLEAN_PARAM(ForReadBalance);
+FDB_DECLARE_BOOLEAN_PARAM(PreferLowerReadUtil);
 
 struct GetTeamRequest {
-	bool wantsNewServers;
+	bool wantsNewServers; // In additional to servers in completeSources, try to find teams with new server
 	bool wantsTrueBest;
-	bool preferLowerUtilization;
+	bool preferLowerUtilization; // if true, lower utilized team has higher score
 	bool teamMustHaveShards;
+	bool forReadBalance;
+	bool preferLowerReadUtil; // only make sense when forReadBalance is true
 	double inflightPenalty;
 	std::vector<UID> completeSources;
 	std::vector<UID> src;
 	Promise<std::pair<Optional<Reference<IDataDistributionTeam>>, bool>> reply;
 
+	typedef Reference<IDataDistributionTeam> TeamRef;
+
 	GetTeamRequest() {}
 	GetTeamRequest(WantNewServers wantsNewServers,
 	               WantTrueBest wantsTrueBest,
 	               PreferLowerUtilization preferLowerUtilization,
 	               TeamMustHaveShards teamMustHaveShards,
+	               ForReadBalance forReadBalance = ForReadBalance::False,
+	               PreferLowerReadUtil preferLowerReadUtil = PreferLowerReadUtil::False,
 	               double inflightPenalty = 1.0)
 	  : wantsNewServers(wantsNewServers), wantsTrueBest(wantsTrueBest), preferLowerUtilization(preferLowerUtilization),
-	    teamMustHaveShards(teamMustHaveShards), inflightPenalty(inflightPenalty) {}
+	    teamMustHaveShards(teamMustHaveShards), forReadBalance(forReadBalance),
+	    preferLowerReadUtil(preferLowerReadUtil), inflightPenalty(inflightPenalty) {}
+
+	// return true if a.score < b.score
+	[[nodiscard]] bool lessCompare(TeamRef a, TeamRef b, int64_t aLoadBytes, int64_t bLoadBytes) const {
+		int res = 0;
+		if (forReadBalance) {
+			res = preferLowerReadUtil ? greaterReadLoad(a, b) : lessReadLoad(a, b);
+		}
+		return res == 0 ? lessCompareByLoad(aLoadBytes, bLoadBytes) : res < 0;
+	}
+
+	// return true if preferHigherUtil && aLoadBytes <= bLoadBytes (higher load bytes has larger score)
+	// or preferLowerUtil && aLoadBytes > bLoadBytes
+	bool lessCompareByLoad(int64_t aLoadBytes, int64_t bLoadBytes) const {
```

Comment:
This method and several similar methods in `GetTeamRequest` should be private, because they're only used by `GetTeamRequest::lessCompare`.